core/src/main/java/org/jruby/RubyArray.java

```diff
@@ -334,35 +340,6 @@ public static RubyArray newArrayNoCopyLight(Ruby runtime, IRubyObject[] args) {
         return arr;
     }
 
-    public static RubyArray newArray(Ruby runtime, Collection<? extends IRubyObject> collection) {
-        // This may seem inefficient for packed arrays, but the cost of this versus is not really worse
-        // than the cost of constructing and walking an Iterator.
-        IRubyObject[] values = collection.toArray(new IRubyObject[collection.size()]);
-        switch (values.length) {
-            case 0: return newEmptyArray(runtime);
-            case 1:
-                if (USE_PACKED_ARRAYS) return new RubyArrayOneObject(runtime, values[0]);
-                break;
-            case 2:
-                if (USE_PACKED_ARRAYS) return new RubyArrayTwoObject(runtime, values[0], values[1]);
-                break;
-        }
-        return new RubyArray(runtime, values);
-    }
-
-    public static RubyArray newArray(Ruby runtime, List<? extends IRubyObject> list) {
```

Comment:
still'd like to have the `List` version around for binary compatibility - it also does not need `toArray` for packed

Reply:
@kares right bringing it back sec :)

Reply:
brought it back here https://github.com/jruby/jruby/pull/4751/files#diff-008115c4ee37b136355736c9aa675fa3R264 with nicely separate packed and not packed paths :)